receiver.go

```diff
@@ -47,36 +47,41 @@ func newReceiver(context *Context, conn *websocket.Conn) *receiver {
 		conn:             conn,
 		context:          context,
 		channel:          make(chan []byte, 10),
+		parseError:       make(chan error, 1),
 		frameDecoder:     getDecompressor(context),
 		pendingRequests:  msgStreamerMap{},
 		pendingResponses: msgStreamerMap{},
 	}
 }
 
 func (r *receiver) receiveLoop() error {
-	defer func() {
-		atomic.AddInt32(&r.activeGoroutines, -1)
-	}()
+	defer atomic.AddInt32(&r.activeGoroutines, -1)
+	atomic.AddInt32(&r.activeGoroutines, 1)
 	go r.parseLoop()
 
-	atomic.AddInt32(&r.activeGoroutines, 1)
+	defer close(r.channel)
+
 	for {
+		if parseErr := errorFromChannel(r.parseError); parseErr != nil {
```

Comment:
As discussed, seems like we can depend on parseLoop fatalError to close, and so don't need check here.